### DISCUSSION TITLE
Marketplace should allow subdirectory actions

### DISCUSSION THREAD

**Discussion by andersk** [community/community#5474] [Actions]
GitHub Actions currently supports aggregating multiple actions as [subdirectories of a single repository](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-using-a-public-action-in-a-subdirectory):

> ### Example: Using a public action in a subdirectory
> `{owner}/{repo}/{path}@{ref}`
>
> A subdirectory in a public GitHub repository at a specific branch, ref, or SHA.
>
> ```yaml
> jobs:
>   my_first_job:
>     steps:
>       - name: My first step
>         uses: actions/aws/ec2@main

This makes it easier to maintain several related actions since their dependencies, build system, test framework, lint setup, etc. can all be shared. GitHub’s own [codeql-action](https://github.com/github/codeql-action) is an example of a repository with several actions (github/codeql-action/init, github/codeql-action/autobuild, github/codeql-action/analyze).

However, it seems subdirectory actions are [not currently allowed in GitHub Marketplace](https://docs.github.com/en/actions/creating-actions/publishing-actions-in-github-marketplace#about-publishing-actions):

> * Each repository must contain a single action.
> * The action's metadata file (`action.yml` or `action.yaml`) must be in the root directory of the repository.

Please consider lifting this restriction.

**Comment by thclark** [comment]
This seems really crazy, we have three actions with closely related code... we're supposed to copy half our code between 3 repos?

**Comment by cortadocodes** [comment]
I too think this is overly restrictive and leads to closely-related code being artificially spread among several repositories instead of one - not a good architecture for my code from a software engineering perspective 😞
